### PR TITLE
Tests validations

### PR DESCRIPTION
**Pull request by jaamanu**
in this PR 
I Add the following validations:
For the User model
For the Post model
Add unit specs for all of your models' methods and validation

### CODE REVIEW

**Comment by codecaiine** [comment]
@jaamanu  You did not set up linters

**Review by codecaiine — COMMENTED** [review]
Hi @jaamanu @kennyegun24 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has not linters set up correctly and please make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/user.rb [hunk 1]
```diff
@@ -2,7 +2,6 @@ class User < ApplicationRecord
   has_many :posts, foreign_key: 'author_id', class_name: 'Post'
```
Comment (mikethreels) [inline app/models/user.rb]: - [ ] the user model should have validations:
- the user can't be blank
- PostCounter should be an integer greater than or equal to zero

File: app/models/post.rb [hunk 2]
```diff
@@ -2,7 +2,6 @@ class Post < ApplicationRecord
   has_many :likes
   has_many :comments
   belongs_to :author, class_name: 'User'
```
Comment (mikethreels) [inline app/models/post.rb:4]: - [ ] Post model should have validations:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi James, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @jaamanu  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**